### PR TITLE
Obstacle Avoidance Assignment

### PR DESCRIPTION
**Pull request by GeorgZs**
As per canvas: Make the SmartCar move forward until it detects an obstacle at a distance of your choosing.

Car drives - ✅ 
Car detects obstacle - ✅ 
Car stops due to obstacle proximity - ✅

### CODE REVIEW

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,360 @@
+{
```
Comment (platisd) [inline .vscode/c_cpp_properties.json:1]: This file shouldn't be version controlled

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -1,9 +1,44 @@
...
+   * @brief if the distance is greater than 0 and less than 35
+   * meaning it has picked up an obstacle but waits until it has reached
+   * a point before crashing, then it stops.
+   *
+   * Value for min distance to obstalce = 35;
+   * 
+   * @else drive normally
+   * 
+   */
+  if(frontUltraSonic.getDistance() > 0 && frontUltraSonic.getDistance() > 35){
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: This doesn't look right :sweat_smile:

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (platisd) [inline .vscode/settings.json:1]: This file shouldn't be version controlled

**Review by platisd — CHANGES_REQUESTED** [review]

File: .vscode/c_cpp_properties.json (hunk 1 above)
Comment (GeorgZs) [reply .vscode/c_cpp_properties.json:1]: Without these I do not have access to the smartcar library, its a weird mechanism that VS Code uses

Comment (platisd) [reply .vscode/c_cpp_properties.json:1]: Sure, you can keep them but you shouldn't version control them. They even contain paths to **your** machine.

Comment (GeorgZs) [reply .vscode/c_cpp_properties.json:1]: Understood, will remove them in future pull requests! Thanks for the heads up

**Pull request closed without merging**